vars/mavenCanaryRelease.groovy

```diff
@@ -65,4 +70,159 @@ def call(body) {
     }
 
     contentRepository(body);
-  }
+}
+
+def patchFMPVersion() {
+    def pomModel = readMavenPom file: 'pom.xml'
+
+    if (!pomModel.profiles) {
+        return false
+    }
+
+    def updatedPlugin = false
+    for(profile in pomModel.profiles) {
+        if(!profile.id.equalsIgnoreCase("openshift") && !profile.build.plugins) {
+            continue
+        }
+
+        for(plugin in profile.build.plugins) {
+            if(plugin.artifactId != "fabric8-maven-plugin") {
+                continue
+            }
+
+            if(plugin.version) {
+                if (compareVersions(plugin.version, "3.5.38") < 0) {
+                    println "patching maven plugin for fabric8-maven-plugin v3.5.38"
+                    plugin.version = "3.5.38"
+                    updatedPlugin = true
+                } else {
+                    return false
+                }
+            } else {
+                updatedPlugin = patchBasedOnParentForPlugin(plugin, pomModel.parent)
+            }
+        }
+    }
+
+    // there is no FMP plugin found and then update FMP based on parent module definition
+    if(!updatedPlugin) {
+        updatedPlugin = patchBasedOnParentForProfile(pomModel)
+    }
+
+    if (updatedPlugin) {
+        writeMavenPom model: pomModel
+    }
+
+    return updatedPlugin
+}
+
+/*
+* if v1 is greater than v2 returns false
+* if v2 is greater than v1 returns true
+* if v1 is equal to v2 return false
+*/
+def compareVersions(v1, v2) {
+    (v1Parts, v2Parts) = partitionVersions(v1, v2)
+    println v1Parts
+    println v2Parts
+
+    for (int i=0; i < maxParts; i++){
+        if(v1Parts[i] != v2Parts[i]){
+            return v1Parts[i] - v2Parts[i]
+        }
+    }
+
+    return 0
+}
+
+def zeroFill(v, count) {
+    (0..count).each {
+        v << 0
+    }
+}
+
+/**
+ * given versions strings v1 and v2 returns versions partitioned into
+ * an array of strings of equal length. e.g.
+ * "1.2", "3.4.5" returns [1, 2, 0], [3, 4, 5]
+ * "1.2.3-5", "3.4.5" returns [1, 2, 3, 15], [3, 4, 5]
+ */
+def partitionVersions(v1, v2) {
+    delim = '[\\.-]'
```

Comment:
@hrishin just learned that without `def` the variable becomes something  close to a global variable; so we have to add `def` here for all var declarations. e.g. try this

```groovy

def f() {
  x = 1
}

f()
println x
```